relay-event-schema/src/protocol/span/convert.rs

```diff
@@ -129,38 +135,30 @@ macro_rules! map_fields {
 // This macro call implements a bidirectional mapping between transaction event and segment spans,
 // allowing users to call both `Event::from(&span)` and `Span::from(&event)`.
 map_fields!(
-    top_level {
-        span._metrics_summary <=> event._metrics_summary,
-        span.description <=> event.transaction,
-        span.measurements <=> event.measurements,
-        span.platform <=> event.platform,
-        span.received <=> event.received,
-        span.start_timestamp <=> event.start_timestamp,
-        span.tags <=> event.tags,
-        span.timestamp <=> event.timestamp
-    }
-    contexts {
-        TraceContext {
-            span.exclusive_time <=> context.exclusive_time,
-            span.op <=> context.op,
-            span.parent_span_id <=> context.parent_span_id,
-            // A transaction corresponds to a segment span, so span_id and segment_id have the same value:
-            span.span_id, span.segment_id <=> context.span_id,
-            span.status <=> context.status,
-            span.trace_id <=> context.trace_id
-        }
-        ProfileContext {
-            span.profile_id <=> context.profile_id
-        }
-    }
-    fixed_for_span {
-        // A transaction event corresponds to a segment span.
-        span.is_segment <= true,
-        span.was_transaction <= true
-    }
-    fixed_for_event {
-        EventType::Transaction => event.ty
-    }
+    span._metrics_summary <=> event._metrics_summary,
+    span.description <=> event.transaction,
+    span.measurements <=> event.measurements,
+    span.platform <=> event.platform,
+    span.received <=> event.received,
+    span.start_timestamp <=> event.start_timestamp,
+    span.tags <=> event.tags,
+    span.timestamp <=> event.timestamp,
+    span.exclusive_time <=> event.contexts.trace.exclusive_time,
+    span.op <=> event.contexts.trace.op,
+    span.parent_span_id <=> event.contexts.trace.parent_span_id,
+    // A transaction corresponds to a segment span, so span_id and segment_id have the same value:
+    span.span_id <=> event.contexts.trace.span_id,
+    span.segment_id <=> event.contexts.trace.span_id,
+    span.status <=> event.contexts.trace.status,
+    span.trace_id <=> event.contexts.trace.trace_id,
+    span.profile_id <=> event.contexts.profile.profile_id,
+    span.data.release <=> event.release,
+    span.data.environment <=> event.environment
+    ;
+    span.is_segment <= true,
+    span.was_transaction <= true
+    ;
+    EventType::Transaction => event.ty
```

Comment:
`map_fields!` is now called with a flat list of field correspondences, as I intended it initially.